packages/taquito/src/contract/interface.ts

```diff
@@ -55,8 +61,19 @@ export interface EstimationProvider {
    *
    * @param Estimate
    */
-  registerDelegate(params?: RegisterDelegateParams): Promise<Estimate>;
-  batch(params: ParamsWithKind[]): Promise<Estimate[]>;
+  registerDelegate(params?: RegisterDelegateParams, estimationOptions?: EstimationOptions): Promise<Estimate>;
+
+  /**
+   *
+   * @description Estimate gasLimit, storageLimit and fees for a reveal operation
+   *
+   * @returns An estimation of gasLimit, storageLimit and fees for the operation or undefined if the account is already revealed
```

Comment:
What if we want to estimate a reveal operation after having an account revealed?

Reply:
This would result in an error from the RPC saying that the account is already revealed when simulating the operation.